### PR TITLE
Row gallery enhancements

### PR DESCRIPTION
**Pull request by uldisrudzitis**
Load from server limited gallery items and generate only remaining items client side.

row-gallery-loader

### CODE REVIEW

File: engine/js/BertaGalleryRow.js [hunk 1]
```diff
@@ -169,101 +170,88 @@ var BertaGalleryRow = new Class({
     liElement.addClass('selected');
   },
 
-  load: function (src, mType, mWidth, mHeight, videoPath, autoPlay, caption, bDeleteExisting, xImgIndex, srcset) {
+  load: function (src, mType, mWidth, mHeight, videoPath, autoPlay, caption, xImgIndex, srcset) {
+    this.currentItem += 1;
+    this.itemIsLoaded = this.currentItem <= this.loadedItems;
```
Comment (vteivans) [inline engine/js/BertaGalleryRow.js]: let's call this `currentItemIsLoaded`. Otherwise, it's not clear, what is the loaded item since the gallery class handles all of the items.

File: engine/js/BertaGalleryRow.js [hunk 2]
```diff
@@ -198,114 +170,88 @@ var BertaGalleryRow = new Class({
...
-        this.imageShowFx.cancel();
-        break;
-    }
-
+  load: function (src, mType, mWidth, mHeight, videoPath, autoPlay, caption, xImgIndex, srcset) {
+    this.currentItem += 1;
+    this.itemIsLoaded = this.currentItem <= this.loadedItems;
     this.currentSrc = null;
-    this.load_Render(src, mType, mWidth, mHeight, videoPath, autoPlay, caption, bDeleteExisting, xImgIndex, srcset);
+    this.load_Render(src, mType, mWidth, mHeight, videoPath, autoPlay, caption, xImgIndex, srcset);
```
Comment (vteivans) [inline engine/js/BertaGalleryRow.js]: I see that there are a lot of parameters to this function, but maybe we should pass the `isLoaded` parameter as well? I think it could make the code a bit easier to follow. And it doesn't seem like, it's used in lot of functions around.

File: engine/_classes/class.bertagallery.php [hunk 3]
```diff
@@ -135,7 +131,9 @@ public static function getHTML($imgs, $mediaFolderName, $galleryType, $isAdminMo
                     $spaceBetweenItems = 12;
                     $totalWidth += $itemWidth + $spaceBetweenItems;
 
-                    $galleryContent .= $itemHTML;
+                    if ($i < $imageLimit) {
```
Comment (vteivans) [inline engine/_classes/class.bertagallery.php]: why did you change the if statement? Wasn't it better before?

Comment (uldisrudzitis) [reply engine/_classes/class.bertagallery.php]: Because we can't break the loop as before. We need to loop through all the items to get the total width of gallery.

Comment (vteivans) [reply engine/_classes/class.bertagallery.php]: Also doesn't the navigation part already loop over all the images?

Comment (uldisrudzitis) [reply engine/_classes/class.bertagallery.php]: No, it generates only the limited images, see
```
if ($i < $imageLimit) {
  $galleryContent .= $itemHTML;
}
```

all the other images are loaded in frontend based on the gallery navigation html code
